x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -49,13 +65,32 @@ export const hostMiddlewareFactory: ImmutableMiddlewareFactory<HostState> = (cor
           });
       }
 
-      const { page_index: pageIndex, page_size: pageSize } = uiQueryParams(state);
+      const { page_index: pageIndex, page_size: pageSize, admin_query: adminQuery } = uiQueryParams(
+        state
+      );
       let hostResponse;
 
       try {
+        // get index pattern and fields for search bar
+        fetchIndexPatterns()
+          .then((patterns) => {
+            dispatch({
+              type: 'serverReturnedMetadataPatterns',
+              payload: patterns,
+            });
+          })
+          .catch((error) => {
+            // what to do with error??
+            dispatch({
+              type: 'serverFailedToReturnMetadataPatterns',
+              payload: error,
+            });
+          });
+
         hostResponse = await coreStart.http.post<HostResultList>('/api/endpoint/metadata', {
           body: JSON.stringify({
             paging_properties: [{ page_index: pageIndex }, { page_size: pageSize }],
+            filter: adminQuery ? ((decode(adminQuery) as unknown) as Query).query : '',
```

Comment:
FYI this PR: https://github.com/elastic/kibana/pull/71882

changes changes the schema to now be:

```
body: {
    paging_properties: ...,
    filters: {
        kql: ...
    }
}
```

Since we are likely adding additional kinds of filters.